### PR TITLE
cm3: scb: add Cortex-M7 cache registers and bits

### PR DESCRIPTION
**Pull request by j4cbo**
Cortex-M7 supports a D-cache and I-cache on the AXI bus, controlled by these bits in SCB.

### CODE REVIEW

File: include/libopencm3/cm3/scb.h [hunk 1]
```diff
@@ -320,6 +365,17 @@
...
 #endif
+
+/* Those defined only on ARMv7EM and above */
+#if defined(__ARM_ARCH_7EM__)
+/** BP set to enable branch predictor */
+#define SCB_CCR_BP					(1 << 18)
+/** IC set to enable instruction cache */
+#define SCB_CCR_IC					(1 << 17)
+/** DC set to enable data cache */
+#define SCB_CCR_DC					(1 << 17)
```
Comment (karlp) [inline include/libopencm3/cm3/scb.h:376]: are you sure about these bits? you've got 17 twice?

File: include/libopencm3/cm3/scb.h [hunk 2]
```diff
@@ -443,6 +499,26 @@
...
+#define SCB_CTR_CWG_MASK		(0xf << SCB_CTR_CWG_SHIFT)
+/* ERG: Exclusives Reservation Granule */
+#define SCB_CTR_ERG_SHIFT		20
+#define SCB_CTR_ERG_MASK		(0xf << SCB_CTR_ERG_SHIFT)
+/* DMINLINE: log2 of number of words in smallest cache line of all data caches */
+#define SCB_CTR_DMINLINE_SHIFT	16
+#define SCB_CTR_DMINLINE_MASK	(0x1f << SCB_CTR_DMINLINE_SHIFT)
+/* IMINLINE: log2 of number of words in smallest cache line of all instruction caches */
+#define SCB_CTR_IMINLINE_SHIFT	0
+#define SCB_CTR_IMINLINE_MASK	(0xf << SCB_CTR_IMINLINE_SHIFT)
```
Comment (karlp) [inline include/libopencm3/cm3/scb.h:518]: generally we provide masks unshifted.

File: include/libopencm3/cm3/scb.h (hunk 1 above)
Comment (j4cbo) [reply include/libopencm3/cm3/scb.h]: oops! fixed

File: include/libopencm3/cm3/scb.h (hunk 2 above)
Comment (j4cbo) [reply include/libopencm3/cm3/scb.h]: done

**Comment by karlp** [comment]
merged in 5b9b784b thanks!

**Pull request closed without merging**